src/teleop/gui/src/components/AutonWaypointEditor.vue

```diff
@@ -433,42 +433,53 @@ export default {
   },
 
   beforeDestroy: function () {
-    window.clearInterval(interval);
+    window.clearInterval(stuck_interval);
+    window.clearInterval(intermediate_publish_interval);
     this.autonEnabled = false;
     this.sendEnableAuton();
   },
 
   created: function () {
-    (this.course_pub = new ROSLIB.Service({
+    this.course_pub = new ROSLIB.Topic({
       ros: this.$ros,
-      name: "/enable_auton",
-      serviceType: "mrover/PublishEnableAuton",
-    })),
-      (this.nav_status_sub = new ROSLIB.Topic({
-        ros: this.$ros,
-        name: "/smach/container_status",
-        messageType: "smach_msgs/SmachContainerStatus",
-      })),
-      (this.rover_stuck_pub = new ROSLIB.Topic({
-        ros: this.$ros,
-        name: "/rover_stuck",
-        messageType: "std_msgs/Bool",
-      })),
-      // Make sure local odom format matches vuex odom format
-      (this.odom_format_in = this.odom_format);
+      name: "/intermediate_enable_auton",
+      messageType: "mrover/EnableAuton",
+    });
+
+    this.nav_status_sub = new ROSLIB.Topic({
+      ros: this.$ros,
+      name: "/nav_state",
+      messageType: "std_msgs/String",
+    });
+
+    this.rover_stuck_pub = new ROSLIB.Topic({
+      ros: this.$ros,
+      name: "/rover_stuck",
+      messageType: "std_msgs/Bool",
+    });
+
+    // Make sure local odom format matches vuex odom format
+    this.odom_format_in = this.odom_format;
 
     this.nav_status_sub.subscribe((msg) => {
-      if (msg.active_states[0] !== "OffState" && !this.autonEnabled) {
+      // If still waiting for nav...
+      if ((msg.data == "OffState" && this.autonEnabled) ||
+          (msg.data !== "OffState" && !this.autonEnabled)) {
         return;
       }
+
       this.waitingForNav = false;
       this.autonButtonColor = this.autonEnabled ? "green" : "red";
     });
 
     // Interval for publishing stuck status for training data
-    interval = window.setInterval(() => {
+    stuck_interval = window.setInterval(() => {
       this.rover_stuck_pub.publish({ data: this.roverStuck });
     }, 100);
+
+    intermediate_publish_interval = window.setInterval(() => {
```

Comment:
Just to make sure I understand, this is being sent on an interval even when auton is disabled because of the risk of the request from the toggleAutonMode being dropped?